teps/0141-platform-context-variables.md

```diff
@@ -221,190 +232,319 @@ The downside of this approach is that Tasks can't easily reference the values of
-file name, for example, "/teps/images/NNNN-workflow.jpg".
--->
+This TEP proposes adding support for the context variable substitutions in TaskRuns and PipelineRuns of the form `$(context.platform.foo)`, where `foo` is the name of a parameter supplied by the platform. Platforms may provide string, array, or object parameters, so TaskRuns and PipelineRuns may also use substitutions like `$(context.platform.my-array-param[*])` or `$(context.platform.my-object-param.key)`.
+
+For example, for the clone + build Pipeline referenced above, a user could define the following PipelineRun:
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  name: clone-kaniko-build-push-run
+spec:
+  context:
+    params:
+    - name: repo_url
+      value: https://github.com/tektoncd/pipeline
+    - name: image_registry
+      value: gcr.io/user
+  pipelineRef:
+    name: clone-kaniko-build-push
+  params:
+  - name: repo-url
+    value: $(context.platform.repo_url)
+  - name: image
+    value: $(context.platform.image_registry)/myapp
+  workspaces:
+  - name: source-code
+    ...
+```
+
+These variables will be supported for Tasks defined inline in TaskRuns or Pipelines defined inline in PipelineRuns and can be used in any Task/Pipeline fields that [currently accept variable substitutions](https://github.com/tektoncd/pipeline/blob/main/docs/variables.md#fields-that-accept-variable-substitutions). These variable substitutions will be supported in referenced remote Tasks and Pipelines as well.
+
+For example, the following example is how to reference the platform variables in remote Pipeline:
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: Pipeline
+metadata:
+  name: clone-kaniko-build-push
+spec:
+  tasks:
+  - name: fetch-source
+    params:
+    - name: repo-url
+      value: $(context.platform.repo_url)
+    taskRef:
+      name: git-clone
+  - name: build-and-push
+  params:
+    - name: image
+      value: $(context.platform.image_registry)/myapp
+    taskRef:
+      name: kaniko-build
+```
+
+**NOTE:** For reusability purpose, the platform-injected parameters must be passed into the parameters of Catalog Tasks and Pipelines. Documentation should be added for Catalog developers.
 
+### RBAC on the Context fields
+
+This TEP proposes adding a new "context" section directly into PipelineRuns and TaskRuns, and implement our own RBAC to ensure that only authorized users can set the "context" field. For example, for the PipelineRun defined above by a PipelineRun author, the platform would submit the following PipelineRun to the Kubernetes cluster:
+
+```yaml
+apiVersion: tekton.dev/v1
+kind: PipelineRun
+metadata:
+  name: clone-kaniko-build-push-run
+spec:
+  pipelineRef:
+    name: clone-kaniko-build-push
+  params:
+  - name: repo-url
+    value: $(context.platform.repo_url)
+  - name: image
+    value: $(context.platform.image_registry)/myapp
+  workspaces:
+  - name: source-code
+    ...
+context:
```

Comment:
are there any additional way of providing `context` instead of or in addition to `pipelineRun.context`?